### DISCUSSION TITLE
is the gem compatible with react-rails?

### DISCUSSION THREAD

**Discussion by killernova** [ElMassimo/vite_ruby#86] [Q&A] [answered]
We have an old project using react-rails gem for Reactjs with webpacker, and we want to move to vite, anyone has experience with vite_ruby and react-rails?

**Comment by ElMassimo** [comment, marked as answer]
Hi there!

I haven't used `react-rails`, but the [server-side rendering functionality](https://github.com/reactjs/react-rails/blob/f8148d57a6d96b27bddfaef9f9bf26e233b27bd6/lib/react/server_rendering/webpacker_manifest_container.rb) seems to depend on webpacker.

The `react_ujs` library it provides seems to rely on either globals or Webpack's [`require.context`](https://github.com/reactjs/react-rails/blob/f8148d57a6d96b27bddfaef9f9bf26e233b27bd6/react_ujs/src/getConstructor/fromRequireContext.js).

If you are __not__ using server-side rendering and are __not__ using `require.context`, then it should be possible for you to migrate your `react-rails` project to Vite.

<hr/>

Otherwise, `react-rails` won't work out of the box with Vite, and requires additional work to integrate with Vite Ruby.

I'd suggest contacting the authors to check if they would be interested—it should be possible to allow passing a [`glob`](https://vitejs.dev/guide/features.html#glob-import) instead of `require.context` to "register" the available components.
